crates/orderbook/src/database/orders.rs

```diff
@@ -498,18 +498,32 @@ impl Postgres {
             .with_label_values(&["token_first_trade_block"])
             .start_timer();
 
-        let mut ex = self.pool.acquire().await?;
-        let block_number = database::trades::token_first_trade_block(&mut ex, ByteArray(token.0))
-            .await
-            .map_err(anyhow::Error::from)?
-            .map(u32::try_from)
-            .transpose()
-            .map_err(anyhow::Error::from)?;
+        let (block_number, native_price) = tokio::join!(
```

Comment:
Given that you return an error if one of the values is missing (returned an error) you can also consider using `try_join` here to cancel both futures if one of them fails.
Not sure if this would make the error handling ever more verbose, though.